### DISCUSSION TITLE
save DeepSleepTime value in memory and update with MQTT

### DISCUSSION THREAD

**Discussion by abhitheawesomecoder** [arendst/Tasmota#16084] [Q&A] [answered]
I want to update deepsleeptime value, which is inside a rule, with mqtt command/request and not to keep deepsleeptime as a constant. I was wondering how to achieve this

**Comment by sfromis** [comment, marked as answer]
You could have it in a Mem variable, persistent across restarts, and update it by publishing a Mem command via MQTT.

**Comment by stefanbode** [comment]
It is more easier if you do it from the outside. Just persist the new deepsleeptime on mqtt and after the next wakeup the device will update its deepsleep. It also works if you externally have a rule on the LWT online and then send the new deepsleeptime.
